File: adidas-ascii-logo/java/src/main/java/com/adidas/AdidasAsciiLogo.java
package com.adidas;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class AdidasAsciiLogo {

    private String repeat(String str, int count) {
        return count > 0 ?  repeat(str, count - 1) + str: "";
    }

    public String createAdidasAsciiLogo(int width) {
        if (width < 2) {
            throw new IllegalArgumentException("minimum width is 2");
        }

        StringBuilder sb = new StringBuilder();
        int numStripes = 3;
        int smallHeight = (int) Math.round(Math.sqrt(width));
        int totalHeight = smallHeight * numStripes;

        // Some vars to speed up things a little bit
        String stripe = this.repeat("@", width);
        String separator = this.repeat(" ", smallHeight);
        List<String> stripes = new ArrayList<String>();
        for (int i = 0; i < numStripes; i++) {
            stripes.add(stripe);
        }

        for (int h = 0; h < totalHeight; h++) {
            int offset = h % smallHeight;
            int skip = numStripes - 1 - (h / smallHeight);

            sb.append(this.repeat(" ", skip * width + offset));
            sb.append(String.join(separator, stripes.subList(skip, numStripes)));

            // Java Test expects carriage return on all lines
            sb.append("\n");
        }

        return sb.toString();
    }
}

File: adidas-ascii-logo/javascript/src/adidas-ascii-logo.js
/**
 * Returns the adidas three stripes logo using `@` characters.
 * @param {number} width - Width of a stripe.
 * @returns {string} adidas logo.
 */
module.exports = function(width) {
  if (width < 2) {
    throw 'Error, minimum width is 2';
  }

  let str = '';
  let numStripes = 3;
  let smallHeight = Math.round(Math.sqrt(width));
  let totalHeight = smallHeight * numStripes

  // Some vars to speed up things a little bit
  let stripe = '@'.repeat(width);
  let separator = ' '.repeat(smallHeight);
  let stripes = Array(numStripes).fill(stripe);

  for (let h = 0; h < totalHeight; h++) {
    let offset = h % smallHeight;
    let skip = numStripes - 1 - Math.floor(h / smallHeight);

    str += ' '.repeat( (skip * width) + offset);
    str += stripes.slice(skip, numStripes).join(separator);

    // Tests expect no last carriage return
    if (h != totalHeight - 1) {
      str += '\n';
    }
  }

  return str;
};
